src/Json.php

```diff
@@ -165,84 +165,107 @@ protected static function recursiveJsonExprFinder(
+                            || ($last === '[' && $json[$i] === ']')
+                        ) {
+                            array_pop($stack);
+                        }
+
+                        $result .= $json[$i];
+                        while (isset($json[$i + 1]) && preg_match('/\s/', $json[$i + 1])) {
+                            ++$i;
+                        }
+                        if (isset($json[$i + 1]) && ($json[$i + 1] === '}' || $json[$i + 1] === ']')) {
+                            $result .= "\n" . str_repeat($indentString, count($stack) - 1);
+                        }
+
+                        continue 2;
+                    }
+                    break;
+                case '"':
+                    $result .= '"';
+
+                    if (! $inLiteral) {
+                        $inLiteral = true;
+                    } else {
+                        $backslashes = 0;
+                        $n = $i;
+                        while ($json[--$n] === '\\') {
+                            ++$backslashes;
+                        }
+
+                        if (($backslashes % 2) === 0) {
+                            $inLiteral = false;
+
+                            while (isset($json[$i + 1]) && preg_match('/\s/', $json[$i + 1])) {
+                                ++$i;
+                            }
+
+                            if (isset($json[$i + 1]) && ($json[$i + 1] === '}' || $json[$i + 1] === ']')) {
+                                $result .= "\n" . str_repeat($indentString, count($stack) - 1);
+                            }
+                        }
+                    }
+                    continue 2;
+                case ':':
+                    if (! $inLiteral) {
+                        $result .= ': ';
+                        continue 2;
+                    }
+                    break;
+                case ',':
+                    if (! $inLiteral) {
+                        $result .= ',' . "\n" . str_repeat($indentString, count($stack));
+                        continue 2;
+                    }
+                    break;
+                default:
+                    if (! $inLiteral && preg_match('/\s/', $json[$i])) {
+                        continue 2;
+                    }
+                    break;
             }
 
-            if (preg_match('/^("(?:.*)"):[ ]?(.*)$/', $token, $matches)) {
-                $token = $matches[1] . ': ' . $matches[2];
-            }
+            $result .= $json[$i];
 
-            $prefix = str_repeat($indentString, $indentLevel);
-            if (! $inLiteral && in_array($token, $openingBrackets, true)) {
-                $indentLevel++;
-                if ($result != '' && $result[strlen($result) - 1] === "\n") {
-                    $result .= $prefix;
-                }
-                $result .= $token;
-
-                $closingBracket = $bracketPairs[$token];
-
-                do {
+            if (! $inLiteral) {
```

Comment:
Here, I would return early:

```php
if ($inLiteral) {
    return;
}
// promote the conditional expression up a level here
```

Reply:
I can't as this is inside `for` loop.

Reply:
So do a `continue` instead of a `return`. :)

Reply:
Updated, but honestly I am not sure if it is clearer.

Reply:
I just viewed the file as committed (i.e., not the diff, but the result following diff), and it is much clearer to me that if we are in a literal and have not hit a character that would mark the end of the literal, we advance to the next character and do nothing. So definitely a win!